### PR TITLE
Bug fixes for ML 0.8.0

### PR DESCRIPTION
**Pull request by llali**
There were some issue with pages getting refreshed more than once. Adding a check for make sure the page needs to refresh and also sending "item changed" events when item actually changed helps with not refreshing the pages multiple times 
Also updated the version to 0.8.0 for the new release

### CODE REVIEW

File: extensions/machine-learning/package.json [hunk 1]
```diff
@@ -2,7 +2,7 @@
   "name": "machine-learning",
   "displayName": "%displayName%",
   "description": "%description%",
-  "version": "0.7.0",
+  "version": "0.8.0",
```
Comment (udeeshagautam) [inline extensions/machine-learning/package.json:5]: dont we need to update ADS depepndency version also?

File: extensions/machine-learning/src/views/wizardView.ts [hunk 2]
```diff
@@ -112,12 +112,6 @@ export class WizardView extends MainViewBase {
 	}
 
 	public async refresh(): Promise<void> {
-		for (let index = 0; index < this._pages.length; index++) {
```
Comment (udeeshagautam) [inline extensions/machine-learning/src/views/wizardView.ts:115]: when does the pages get added - once and keep that way?

File: extensions/machine-learning/package.json (hunk 1 above)
Comment (llali) [reply extensions/machine-learning/package.json:5]: sure I'll do that

---
In reply to: [592654459](https://github.com/microsoft/azuredatastudio/pull/14659#discussion_r592654459) [](ancestors = 592654459)

File: extensions/machine-learning/src/views/wizardView.ts (hunk 2 above)
Comment (llali) [reply extensions/machine-learning/src/views/wizardView.ts:115]: pages are added when wizard is created, I added this code so if after wizard is created I can add new pages. It's causing issues and I don't need it in any wizard right now 

---
In reply to: [592657209](https://github.com/microsoft/azuredatastudio/pull/14659#discussion_r592657209) [](ancestors = 592657209)

File: extensions/machine-learning/package.json (hunk 1 above)
Comment (Charles-Gagnon) [reply extensions/machine-learning/package.json:5]: Do you need to though? You only need to update the version if there's new changes in core that you require for the extension to function. These fixes all seem to be in the extension so shouldn't require updating the engine version.

Comment (llali) [reply extensions/machine-learning/package.json:5]: yes but there were changes in core for some UI work, already in RC which this version of extension depends on it 

---
In reply to: [592664355](https://github.com/microsoft/azuredatastudio/pull/14659#discussion_r592664355) [](ancestors = 592664355)

Comment (Charles-Gagnon) [reply extensions/machine-learning/package.json:5]: Cool, make sure you update the field in the gallery files when you update those as well

**Review by udeeshagautam — APPROVED** [review]

**Pull request merged**